lib/CsvImportValidator.class.php

```diff
@@ -0,0 +1,466 @@
+        'QubitAccession',
+        'QubitRepository',
+        'QubitEvent',
+    ];
+
+    // General tests which can apply to any CSV.
+    protected $generalTestList = [
+        'CsvSampleValuesValidator' => CsvSampleValuesValidator::class,
+        'CsvFileEncodingValidator' => CsvFileEncodingValidator::class,
+        'CsvColumnNameValidator' => CsvColumnNameValidator::class,
+        'CsvColumnCountValidator' => CsvColumnCountValidator::class,
+        'CsvDuplicateColumnNameValidator' => CsvDuplicateColumnNameValidator::class,
+        'CsvEmptyRowValidator' => CsvEmptyRowValidator::class,
+        'CsvCultureValidator' => CsvCultureValidator::class,
+        'CsvLanguageValidator' => CsvLanguageValidator::class,
+        'CsvFieldLengthValidator' => CsvFieldLengthValidator::class,
+    ];
+
+    // Tests which pertain only to this class type.
+    protected $qubitInformationObjectTestList = [
+        'CsvParentValidator' => CsvParentValidator::class,
+        'CsvLegacyIdValidator' => CsvLegacyIdValidator::class,
+        'CsvEventValuesValidator' => CsvEventValuesValidator::class,
+        'CsvDigitalObjectPathValidator' => CsvDigitalObjectPathValidator::class,
+        'CsvDigitalObjectUriValidator' => CsvDigitalObjectUriValidator::class,
+    ];
+
+    public function __construct(sfContext $context = null,
+    $dbcon = null,
+    $options = [])
+    {
+        if (null === $context) {
+            $context = new sfContext(ProjectConfiguration::getActive());
+        }
+
+        $this->setContext($context);
+        $this->dbcon = $dbcon;
+        $this->setOptions($options);
+
+        $this->setOrmClasses(
+            [
+                'QubitFlatfileImport' => QubitFlatfileImport::class,
+                'QubitObject' => QubitObject::class,
+            ]
+        );
+    }
+
+    public function getTestsByClassType()
+    {
+        $selectedTests = $this->generalTestList;
+
+        switch ($this->validatorOptions['className']) {
+            case 'QubitInformationObject':
+                $selectedTests = array_merge($selectedTests, $this->qubitInformationObjectTestList);
+
+                break;
+        }
+
+        return $selectedTests;
+    }
+
+    public function loadCsvData($fh)
+    {
+        $this->handleByteOrderMark($fh);
+        $this->header = fgetcsv($fh, 60000, $this->getOption('separator'), $this->getOption('enclosure'));
+
+        if (false === $this->header) {
+            throw new sfException('Could not read initial row. File could be empty.');
+        }
+
+        $this->rows = [];
+        while ($item = fgetcsv($fh, 60000, $this->getOption('separator'), $this->getOption('enclosure'))) {
+            $this->rows[] = $item;
+        }
+
+        // Remove trailing blank lines. Iterate over array from bottom up,
+        // removing empty rows.
+        for ($i = count($this->rows) - 1; $i >= 0; --$i) {
+            if (!(1 == count($this->rows[$i]) && empty(trim(implode('', $this->rows[$i]))))) {
```

Comment:
I find this conditional really confusing.  

I think it would be clearer without the initial "not":

```
if (1 !== count($this->rows[$i]) || empty(trim(implode('', $this->rows[$i])))) {
    return;
}
```

I also don't think the `implode()` is necessary, because the the first condition ensures that there is one and only one element in the array.

Reply:
Will clean up that initial 'not'. The implode() is necessary though because even when there is only one column, $this->rows[$i] is an array.